eng/packaging.targets

```diff
@@ -42,4 +54,76 @@
     <MSBuild Projects="$(PkgProjPath)"
              Targets="Build" />
   </Target>
+    
+  <Target Name="AddRefAssemblies" Condition="'$(IncludeRefInPackage)' == 'true' and '$(TargetsAnyOS)' == 'true' and '@(ResolvedMatchingContract)' != ''"> 
+    <ItemGroup>
+      <TfmSpecificPackageFile Include="@(ResolvedMatchingContract)">
+        <RefAssemblyOutput>true</RefAssemblyOutput>
+        <PackagePath>ref/$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))</PackagePath>
+        <PackagePathWithFilename>ref/$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))/%(ResolvedMatchingContract.Filename).dll</PackagePathWithFilename>
+      </TfmSpecificPackageFile>
+      <TfmSpecificPackageFile Include="$(XmlDocFileRoot)1033\$(TargetName).xml" Condition="Exists('$(XmlDocFileRoot)1033\$(TargetName).xml')">
+        <RefAssemblyOutput>true</RefAssemblyOutput>
+        <PackagePath>ref/$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))</PackagePath>
+        <PackagePathWithFilename>ref/$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))/%(ResolvedMatchingContract.Filename).xml</PackagePathWithFilename>
+      </TfmSpecificPackageFile>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AddRuntimeSpecificAssemblies"
+          DependsOnTargets="BuiltProjectOutputGroup;SatelliteDllsProjectOutputGroup;_AddPriFileToPackBuildOutput;$(TargetsForTfmSpecificBuildOutput)"
+          Condition="'$(TargetsAnyOS)' != 'true' and '$(ExcludeFromPackage)' == ''">
+    <ItemGroup>
+      <TfmSpecificPackageFile Include="@(SatelliteDllsProjectOutputGroupOutput);
+                            @(BuiltProjectOutputGroupOutput);
+                            @(_PathToPriFile)">
+        <PackagePath>runtimes/$(_runtimeOS)/lib/$(TargetFrameworkWithoutSuffix)</PackagePath>
+      </TfmSpecificPackageFile>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AddRefDebugSymbolsWithTfm"
+          Condition="'$(IncludeSymbols)' == 'true' and '$(IncludeRefInPackage)' == 'true' and '$(TargetsAnyOS)' == 'true'">
+    <PropertyGroup>
+      <RefSymbolPath>$(BaseOutputPath)ref\$(TargetFrameworkWithoutSuffix)-$(Configuration)\$(TargetName).pdb</RefSymbolPath>
```

Comment:
Does this run as part of the reference source project or the source project? If the former, could we use `$(TargetPath)` instead as in another comment and change the extension?

Reply:
This runs as  a part of src project.